packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -341,52 +386,59 @@ function DashboardLayout(props: DashboardLayoutProps) {
     >
       <AppBar color="inherit" position="absolute" sx={{ displayPrint: 'none' }}>
         <Toolbar sx={{ backgroundColor: 'inherit', mx: { xs: -0.75, sm: -1 } }}>
-          <Stack
-            direction="row"
-            justifyContent="space-between"
-            alignItems="center"
-            sx={{
-              flexWrap: 'wrap',
-              width: '100%',
-            }}
-          >
-            <Stack direction="row">
-              {!hideNavigation ? (
-                <React.Fragment>
-                  <Box
-                    sx={{
-                      mr: { sm: disableCollapsibleSidebar ? 0 : 1 },
-                      display: { md: 'none' },
-                    }}
-                  >
-                    {getMenuIcon(isMobileNavigationExpanded)}
-                  </Box>
-                  <Box
-                    sx={{
-                      display: { xs: 'none', md: disableCollapsibleSidebar ? 'none' : 'block' },
-                      mr: disableCollapsibleSidebar ? 0 : 1,
-                    }}
-                  >
-                    {getMenuIcon(isDesktopNavigationExpanded)}
-                  </Box>
-                </React.Fragment>
-              ) : null}
-              {slots?.appTitle ? (
-                <slots.appTitle {...slotProps?.appTitle} />
-              ) : (
-                /* Hierarchy of application of `branding`
-                 * 1. Branding prop passed in the `slotProps.appTitle`
-                 * 2. Branding prop passed to the `DashboardLayout`
-                 * 3. Branding prop passed to the `AppProvider`
-                 */
-                <AppTitle branding={branding} {...slotProps?.appTitle} />
-              )}
-            </Stack>
-            <Stack direction="row" alignItems="center" spacing={1} sx={{ marginLeft: 'auto' }}>
-              <ToolbarActionsSlot {...slotProps?.toolbarActions} />
-              <ToolbarAccountSlot {...slotProps?.toolbarAccount} />
+          {slots?.toolbar ? (
```

Comment:
I would not add a new slot for now, as we still haven't decided how customizations options for the header as a whole will evolve.

Reply:
1. The appBar slot is designed to give users full control over the entire top bar
2. When using the appBar slot, users can:
- Completely customize the layout
- Add their own menu icon
- Handle their own collapse/expand logic
- Place elements wherever they want